testing/endtoend/evaluators/beaconapi_evaluators/beacon_api.go

```diff
@@ -0,0 +1,440 @@
+				blockP,
+				blockL)
+		}
+	} else {
+		blockP := &ethpb.SignedBeaconBlockBellatrix{}
+		blockL := &ethpb.SignedBeaconBlockBellatrix{}
+		if err := blockL.UnmarshalSSZ(sszrspL); err != nil {
+			return errors.Wrap(err, "lighthouse ssz error")
+		}
+		if err := blockP.UnmarshalSSZ(sszrspP); err != nil {
+			return errors.Wrap(err, "prysm ssz error")
+		}
+
+		if len(blockP.Signature) == 0 || len(blockL.Signature) == 0 || hexutil.Encode(blockP.Signature) != hexutil.Encode(blockL.Signature) {
+			return fmt.Errorf("prysm response %v does not match lighthouse response %v",
+				blockP,
+				blockL)
+		}
+	}
+	blockheaderData := beaconPathsAndObjects["/beacon/headers/{param1}"]
+	prysmHeader, ok := blockheaderData.prysmResps["json"].(*apimiddleware.BlockHeaderResponseJson)
+	if !ok {
+		return errors.New("failed to cast type")
+	}
+	proposerdutiesData := beaconPathsAndObjects["/validator/duties/proposer/{param1}"]
+	prysmDuties, ok := proposerdutiesData.prysmResps["json"].(*apimiddleware.ProposerDutiesResponseJson)
+	if !ok {
+		return errors.New("failed to cast type")
+	}
+	if prysmHeader.Data.Root != prysmDuties.DependentRoot {
+		fmt.Printf("current slot: %v\n", slots.CurrentSlot(uint64(genesisData.Data.GenesisTime.AsTime().Unix())))
+		return fmt.Errorf("header root %s does not match duties root %s ", prysmHeader.Data.Root, prysmDuties.DependentRoot)
+	}
+
+	return nil
+}
+
+func compareJSONMulticlient(beaconNodeIdx int, base string, path string, respJSONPrysm interface{}, respJSONLighthouse interface{}, customEvaluator func(interface{}, interface{}) error) error {
+	if err := doMiddlewareJSONGetRequest(
+		base,
+		path,
+		beaconNodeIdx,
+		respJSONPrysm,
+	); err != nil {
+		return errors.Wrap(err, "prysm json error")
+	}
+
+	if err := doMiddlewareJSONGetRequest(
+		base,
+		path,
+		beaconNodeIdx,
+		respJSONLighthouse,
+		"lighthouse",
+	); err != nil {
+		return errors.Wrap(err, "lighthouse json error")
+	}
+	if customEvaluator != nil {
+		return customEvaluator(respJSONPrysm, respJSONLighthouse)
+	} else {
+		return compareResponseObjects(respJSONPrysm, respJSONLighthouse)
+	}
+}
+
+func compareSSZMulticlient(beaconNodeIdx int, base string, path string) ([]byte, []byte, error) {
+	sszrspL, err := doMiddlewareSSZGetRequest(
+		base,
+		path,
+		beaconNodeIdx,
+		"lighthouse",
+	)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "lighthouse json error")
+	}
+
+	sszrspP, err := doMiddlewareSSZGetRequest(
+		base,
+		path,
+		beaconNodeIdx,
+	)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "prysm json error")
+	}
+	if !bytes.Equal(sszrspL, sszrspP) {
+		return nil, nil, fmt.Errorf("prysm ssz response %s does not match lighthouse ssz response %s",
+			hexutil.Encode(sszrspP),
+			hexutil.Encode(sszrspL))
+	}
+	fmt.Printf("prysm ssz: %v", sszrspP)
+	fmt.Printf("lighthouse ssz: %v", sszrspL)
+	return sszrspP, sszrspL, nil
+}
+
+func compareResponseObjects(prysmResp interface{}, lighthouseResp interface{}) error {
```

Comment:
Maybe change the name to `compareJSONObjects` to inform this should not be used for other encodings